src/wizard-settings/header-bidding-gam/index.js

```diff
@@ -224,56 +124,125 @@ const HeaderBiddingGAM = () => {
+										description={ () => (
+											<span>
+												{ sprintf(
+													// Translators: the bidder revenue share for this order.
+													__( 'Bidder Revenue Share: %1$d%%', 'newspack-ads' ),
+													order.revenue_share || 0
+												) }{ ' ' }
+												|{ ' ' }
+												{ sprintf(
+													// Translators: comma-separated list of adapters for the order or "any" if undefined.
+													__( 'Bidders: %s', 'newspack-ads' ),
+													order.bidders?.length
+														? order.bidders
+																.map( bidderKey => bidders[ bidderKey ]?.name || bidderKey )
+																.join( ', ' )
+														: __( 'any', 'newspack-ads' )
+												) }
+											</span>
+										) }
+										actionText={
+											<OrderPopover
+												isDraft={ order.status === 'DRAFT' }
+												disabled={ inFlight }
+												onArchive={ async () => {
+													if (
+														// eslint-disable-next-line no-alert
+														confirm(
+															__( "Are you sure you'd like to archive this order?", 'newspack-ads' )
+														)
+													) {
+														setInFlight( true );
+														await archiveOrder( order.id );
+														setInFlight( false );
+														await fetchOrders();
+													}
+												} }
+												onEdit={ () => setEditingOrder( order.id ) }
+												gamLink={ getOrderUrl( order.id ) }
+											/>
+										}
+										className="mv0"
+										isSmall
+									/>
+								) ) }
+							</Card>
+
+							{
+								// Display warning if a bidder is being targeted by more than one order.
+								Object.keys( bidders ).map( bidderKey => {
+									const bidderOrders = activeOrders.filter( order =>
+										order.bidders?.includes( bidderKey )
+									);
+									return (
+										bidderOrders.length > 1 && (
+											<Notice
+												key={ bidderKey }
+												isWarning
+												noticeText={ sprintf(
+													// Translators: %1 The name of the bidder. %2 The number of orders for the bidder.
+													__( '%1$s is being targeted by %2$d orders.', 'newspack-ads' ),
+													bidders[ bidderKey ].name,
+													bidderOrders.length
+												) }
+											/>
+										)
+									);
+								} )
+							}
+							<Card buttonsCard noBorder className="justify-end">
+								<Button isSecondary disabled={ inFlight } onClick={ () => setIsManaging( false ) }>
+									{ __( 'Cancel', 'newspack-ads' ) }
+								</Button>
+								<Button isPrimary disabled={ inFlight } onClick={ () => setEditingOrder( 0 ) }>
+									Create new order
+								</Button>
+							</Card>
+						</>
 					) }
+				</Modal>
+			) }
+			{ editingOrder !== false && (
+				<Modal
+					title={
+						editingOrder ? __( 'Edit Order', 'newspack-ads' ) : __( 'Create Order', 'newspack-ads' )
+					}
+					onRequestClose={ () => ! inFlight && setEditingOrder( false ) }
+				>
 					{ unrecoverable && (
```

Comment:
I was able to replicate the issue too, it was missing the reset of the config in the case of an unrecoverable error. Otherwise, the ID of the archived failed order is preserved and the button action attempts to update the archived order instead of creating a new one. ce8038b82a642888410e36eb842b19b152e4289d fixes this behavior.